File: MAP/lab2/src/models/AssignStmt.java
package models;

import utils.IDictionary;



public class AssignStmt implements Statement {
   private String id;
   private IExpression exp;

   public AssignStmt(String _id, IExpression _exp){
       id = _id;
       exp = _exp;
   }


    @Override
    public PrgState execute(PrgState state) {

        IDictionary<String, Integer> dict = state.getDict();
        int val = exp.eval(dict);
        dict.setValue(id, val);
        return state;
    }
    @Override
    public String toString(){
       return id + "=" + exp + "; ";
   }
}

File: MAP/lab7/lab7/Models/AssignStmt.cs
using System.Collections;
using lab7.Utils;

namespace lab7.Models
{
    public class AssignStmt : IStatement
    {
        private IExpression exp;
        private string id;

        public AssignStmt(IExpression exp, string id)
        {
            this.exp = exp;
            this.id = id;
        }

        public PrgState execute(PrgState state)
        {
      
            int result = exp.eval(state.Dict);      
            state.Dict.setValue(id,result);

            return state;

        }

        public override string ToString()
        {
            return "Assign(" + id + "=" + exp + ")";
        }
    }
}